examples/cluster/plot_kmeans_assumptions.py

```diff
@@ -0,0 +1,73 @@
+"""
+====================================
+Demonstration of k-means assumptions
+====================================
+
+This example is meant to illustrate situations where k-means will produce good
```

Comment:
I'm not sure I like talking about good clusters, because that is not very well-defined. Maybe rather say this illustrates properties / assumptions of k-means that might be unexpected or unintuitive?